File: scm-csharp/scheme/primitives/PrimitiveSpecialFolderUserHome.cs
namespace scheme;

public class PrimitiveSpecialFolderUserHome : Primitive
{
    public override string Name()
    {
        return "special-folder-user-home";
    }

    public override string Info()
    {
        return
            "Syntax: (special-folder-user-home)\n" +
            "Library: (scm fs)\n" +
            "Description: Returns the path of the user home directory as a string.\n" +
            "Example:\n" +
            "  (special-folder-user-home) => \"/home/user\"";
    }
    
    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 0, 0);
        var path = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
        return System.IO.Path.TrimEndingDirectorySeparator(path).ToCharArray();
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveSpecialFolderUserHome.java
package scheme.primitives;

import java.io.File;

import scheme.*;

public class PrimitiveSpecialFolderUserHome extends Primitive {
    @Override
    public String name() {
        return "special-folder-user-home";
    }

    @Override
    public String info() {
        return "Syntax: (special-folder-user-home)\n" +
               "Library: (scm fs)\n" +
               "Description: Returns the path of the user home directory as a string.\n" +
               "Example:\n" +
               "  (special-folder-user-home) => \"/home/user\"";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 0, 0);
        return System.getProperty("user.home").toCharArray();
    }
}
